web/src/lib/components/shared-components/map/map.svelte

```diff
@@ -0,0 +1,107 @@
+<script lang="ts">
+  import {
+    MapLibre,
+    GeoJSON,
+    MarkerLayer,
+    AttributionControl,
+    ControlButton,
+    Control,
+    ControlGroup,
+  } from 'svelte-maplibre';
+  import { colorTheme } from '$lib/stores/preferences.store';
+  import { serverConfig } from '$lib/stores/server-config.store';
+  import { MapMarkerResponseDto, api } from '@api';
+  import { onMount } from 'svelte';
+  import Cog from 'svelte-material-icons/Cog.svelte';
+  import type { LngLatLike } from 'maplibre-gl';
+
+  export let mapMarkers: MapMarkerResponseDto[];
+  export let showSettingsModal: boolean | undefined = undefined;
+  export let zoom: number | undefined = undefined;
+  export let center: LngLatLike | undefined = undefined;
+
+  let styleUrl: string;
+
+  onMount(() => {
+    colorTheme.subscribe(handleModeSwitch);
+  });
+
+  async function handleModeSwitch(mode: 'dark' | 'light') {
+    switch (mode) {
+      case 'dark':
+        styleUrl = getDarkStyle();
+        break;
+      case 'light':
+        styleUrl = getLightStyle();
+        break;
+    }
+  }
+
+  function getLightStyle() {
+    return $serverConfig.mapStyles.find((style) => style.theme === 'light')?.url || styleUrl;
+  }
+
+  function getDarkStyle() {
+    return $serverConfig.mapStyles.find((style) => style.theme === 'dark')?.url || styleUrl;
+  }
+</script>
+
+<!-- Re-visit this line later. There is a potential PR that may fix this and allow updates without re-rendering the whole component -->
```

Comment:
Please add a link to the referenced PR here

Reply:
That was more of like a personal note. Actually, that should be fixed in the next days, because there is already a release (https://github.com/dimfeld/svelte-maplibre/releases/tag/v0.5.0). Or I just upgrade manually... 🤔

Reply:
Done ;)